src/Components/Index/ServiceSelectionComponent.tsx

```diff
@@ -33,51 +32,48 @@ import {
 } from '@grafana/ui';
 
 import { SelectFieldButton } from '../Forms/SelectFieldButton';
-import { explorationDS, VAR_DATASOURCE, VAR_FILTERS } from 'services/shared';
-import { map, Observable, Unsubscribable } from 'rxjs';
+import { explorationDS, VAR_DATASOURCE } from 'services/shared';
+import { map, Observable } from 'rxjs';
 import { ByLabelRepeater } from 'Components/ByLabelRepeater';
 import { GrotError } from 'Components/GrotError';
 import { getLiveTailControl, getLokiDatasource } from 'services/scenes';
 import { getFavoriteServicesFromStorage } from 'services/store';
+import { debounce } from 'lodash';
 
 const LIMIT_SERVICES = 20;
 const SERVICE_NAME = 'service_name';
 
 interface ServiceSelectionComponentState extends SceneObjectState {
+  // The body of the component
   body: SceneCSSGridLayout;
-  repeater?: ByLabelRepeater;
-  groupBy: string;
-  metricFn: string;
-
-  showHeading?: boolean;
-  searchQuery?: string;
-  showPreviews?: boolean;
-  topServices?: string[];
-  isTopSeriesLoading: boolean;
+  // We query volume endpoint to get list of all services and order them by volume
+  listOfServicesOrderedByVolume?: string[];
+  // Keeps track of whether service list is being fetched from volume endpoint
+  isListOfServicesLoading: boolean;
+  // Keeps track of the search query in input field
```

Comment:
I have removed  `repeater`, `groupBy`, `metricFn`, `showHeading`, `searchQuery`, `showPreviews` and renamed:
- `topServices` -> `listOfServicesOrderedByVolume`
-  `isTopSeriesLoading` -> `isListOfServicesLoading`
- `topServicesToBeUsed` -> `listOfServicesToQuery`